MonumentAddons.cs

```diff
@@ -4441,6 +4442,68 @@ protected virtual void PostEntitySpawn()
                     spooker.SetFlag(BaseEntity.Flags.Busy, true);
                 }
 
+                var telephone = Entity as Telephone;
+                if (telephone != null && telephone.prefabID == 1009655496)
+                {
+                    string phoneName = null;
+                    var gridCoodr = PhoneController.PositionToGridCoord(telephone.Controller.baseEntity.transform.position);
+
+                    var monumentInfo = Monument.Object as MonumentInfo;
+                    if (monumentInfo != null && !string.IsNullOrEmpty(monumentInfo.displayPhrase.translated))
+                    {
+                        phoneName = monumentInfo.displayPhrase.translated;
+
+                        var monuments = PluginInstance.FindMonumentsByShortName(Monument.ShortName);
```

Comment:
While testing, I realized my current map has 2 harbors (`harbor_1` and `harbor_2`) which both have vanilla telephones. The names are `Harbor AE7` and `Harbor N31`. It's worth noting that the plugin's behavior is different; when spawning a telephone with the plugin, it's auto generated name is `Harbor`. From the standpoint of the plugin, that's correct because the telephone was only spawned at one of the harbors, because the two harbors have completely different layouts.

My concern is that if somebody wanted to do something similar, such as add telephones to multiple different harbors or multiple different fishing villages (there are two small variants both called "Fishing Village"), that players wouldn't be able to differentiate them due to not having the grid coordinates.

To solve that, I think the best option is to hard code specific monument prefabs to always display the grid coordinate, such as for harbors, fishing villages and bunker entrances. My hunch is that vanilla is doing something similar, simply setting `AppendGridToName` property to true for harbor prefabs and others, meaning the grid coordinate probably gets appended to phones at those monuments regardless of whether a given map actually has multiple instances of that monument.

Reply:
I forgot that there are multiple versions of same monument while implementing logic for appending coordinates.

As advised, I hardcoded shortnames of multi-version prefabs into switch. Would list or something else be better approach?

Reply:
Normally I would use an array.